File: src/main/java/hulio13/articlesApi/domain/entity/article/ArticleTitle.java
package hulio13.articlesApi.domain.entity.article;

import hulio13.articlesApi.domain.exception.IllegalStringLengthException;
import jakarta.persistence.AttributeOverride;
import jakarta.persistence.Column;
import jakarta.persistence.Embeddable;
import lombok.AccessLevel;
import lombok.NoArgsConstructor;

@Embeddable
@NoArgsConstructor(force = true, access = AccessLevel.PROTECTED)
@AttributeOverride(name = "Value", column = @Column(name = "title"))
public class ArticleTitle {
    public static final int MAX_LENGTH = 150;

    public final String Value;

    public ArticleTitle(String value){
        if (value.length() > MAX_LENGTH){
            throw new IllegalStringLengthException(
                    "Name length should be less or equal than " + MAX_LENGTH);
        }

        Value = value;
    }
}

File: src/main/java/hulio13/articlesApi/domain/entity/article/ArticleTitle.kt
package hulio13.articlesApi.domain.entity.article

import hulio13.articlesApi.domain.exception.IllegalStringLengthException
import jakarta.persistence.AttributeOverride
import jakarta.persistence.Column
import jakarta.persistence.Embeddable

const val ARTICLES_MAX_LENGTH = 150

@Embeddable
@AttributeOverride(name = "Value", column = Column(name = "title", nullable = false))
data class ArticleTitle(val value: String) : Cloneable {
    init {
        if (value.length > ARTICLES_MAX_LENGTH) {
            throw IllegalStringLengthException(
                "Name length should be less or equal than $ARTICLES_MAX_LENGTH"
            )
        }
    }

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as ArticleTitle

        return value == other.value
    }

    override fun hashCode(): Int {
        return value.hashCode()
    }


}